scripts/ccpp_track_variables.py

```diff
@@ -0,0 +1,220 @@
+                intent = ''
+                for scheme_var in section.variable_list():
+                    exact_match = False
+                    if var == scheme_var.get_prop_value('standard_name'):
+                        logging.debug("Found variable {0} in scheme {1}".format(var,section.title))
+                        found_var=var
+                        exact_match = True
+                        intent = scheme_var.get_prop_value('intent')
+                        break
+                    scheme_var_standard_name = scheme_var.get_prop_value('standard_name')
+                    if scheme_var_standard_name.find(var) != -1:
+                        logging.debug(f"{var} matches {scheme_var_standard_name}")
+                        found_var.append(scheme_var_standard_name)
+                if not found_var:
+                    logging.debug(f"Did not find variable {var} in scheme {section.title}")
+                elif exact_match:
+                    logging.debug(f"Exact match found for variable {var} in scheme {section.title},"
+                                  f" intent {intent}")
+                    #print(f"{var_graph=}")
+                    var_graph.append((section.title,intent))
+                else:
+                    logging.debug(f"Found inexact matches for variable(s) {var} "
+                                  f"in scheme {section.title}:\n{found_var}")
+                    partial_matches[section.title] = found_var
+    if var_graph:
+        logging.debug("Successfully generated variable graph for sdf {0}\n".format(suite.sdf_name))
+    else:
+        success = False
+        logging.error(f"Variable {var} not found in any suites for sdf {suite.sdf_name}\n")
+        if partial_matches:
+            print("Did find partial matches that may be of interest:\n")
+            for key in partial_matches:
+                print("In {0} found variable(s) {1}".format(key, partial_matches[key]))
+
+    return (success,var_graph)
+
+def draw_var_graph(var_graph):
+    """Draw a graphical representation of the variable graph (not yet implemented)"""
+
+    success = True
+
+    return success
+
+    return
+
+def main():
+    """Main routine that traverses a CCPP suite and outputs the list of schemes that modify given variable"""
+
+    (success, sdf, var, configfile, metapath, debug) = parse_arguments(args)
+    if not success:
+        raise Exception('Call to parse_arguments failed.')
+
+    success = setup_logging(debug)
+    if not success:
+        raise Exception('Call to setup_logging failed.')
+
+    (success, suite) = parse_suite(sdf)
+    if not success:
+        raise Exception('Call to parse_suite failed.')
+
+    (success, config) = import_config(configfile, None)
+    if not success:
+        raise Exception('Call to import_config failed.')
+
+    # Variables defined by the host model
+    (success, _, _) = gather_variable_definitions(config['variable_definition_files'], config['typedefs_new_metadata'])
```

Comment:
A comment about this magic could be helpful. I assume given its name, it will update the `config` data structure in place, but it's not obvious. You are also opting (I'm assuming intentionally) to not store the output, which throws a wrench in the obvious nature of this call.

Reply:
I'll be honest, I'm not *100%* sure exactly what this step is doing, but it is necessary for converting metadata from an old format. I've added a comment trying to be as informative as possible.